implementation/src/main/java/io/smallrye/metrics/legacyapi/HistogramAdapter.java

```diff
@@ -55,13 +71,48 @@ public HistogramAdapter register(MpMetadata metadata, MetricDescriptor metricInf
 
             tagsSet.add(Tag.of(LegacyMetricRegistryAdapter.MP_SCOPE_TAG, scope));
 
-            globalCompositeSummary = DistributionSummary.builder(metricInfo.name())
-                    .description(metadata.getDescription())
-                    .baseUnit(metadata.getUnit())
-                    .tags(tagsSet)
-                    .publishPercentiles(0.5, 0.75, 0.95, 0.98, 0.99, 0.999)
-                    .percentilePrecision(PRECISION)
-                    .register(Metrics.globalRegistry);
+            Builder builder = DistributionSummary.builder(metricInfo.name()).description(metadata.getDescription())
+                    .baseUnit(metadata.getUnit()).tags(tagsSet).percentilePrecision(PRECISION);
+
+            if (percentilesConfig != null && percentilesConfig.getValues() != null
+                    && percentilesConfig.getValues().length > 0) {
+                double[] vals = Stream.of(percentilesConfig.getValues()).mapToDouble(Double::doubleValue).toArray();
+                builder = builder.publishPercentiles(vals);
+            } else if (percentilesConfig != null && percentilesConfig.getValues() == null
+                    && percentilesConfig.isDisabled() == true) {
+                //do nothing - percentiles were disabled
+            } else {
+                builder = builder.publishPercentiles(0.5, 0.75, 0.95, 0.98, 0.99, 0.999);
+            }
+
+            if (bucketsConfig != null && bucketsConfig.getValues().length > 0) {
+                double[] vals = Stream.of(bucketsConfig.getValues()).mapToDouble(Double::doubleValue).toArray();
+                builder = builder.serviceLevelObjectives(vals);
+            }
+
+            if (defaultBucketConfig != null && defaultBucketConfig.getIsEnabled() == true) {
```

Comment:
Suggest to use `if (defaultBucketConfig != null && defaultBucketConfig.isEnabled()) {`

Reply:
fixed